db/mysql/ops.go

```diff
@@ -0,0 +1,264 @@
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	users := make([]string, 0)
+	for rows.Next() {
+		var user string
+		if err := rows.Scan(&user); err != nil {
+			return nil, err
+		}
+		users = append(users, user)
+	}
+
+	chatRoom.Users = users
+
+	return &chatRoom, nil
+}
+
+func (c *client) GetChatroomMessages(ctx context.Context, id string) ([]*domain.Message, error) {
+
+	rows, err := c.dbc.Query("SELECT id, text, sender_id, room_id, created_at FROM messages WHERE room_id = ?", id)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	messages := make([]*domain.Message, 0)
+	for rows.Next() {
+		var message domain.Message
+		if err := rows.Scan(&message.ID, &message.Text, &message.SenderID, &message.RoomID, &message.CreatedAt); err != nil {
+			return nil, err
+		}
+		messages = append(messages, &message)
+	}
+
+	return messages, nil
+}
+
+func (c *client) createChatRoom(name string) error {
+
+	// Ensure chat room name is not blank
+	if name == "" {
+		return errors.New("Chat room name cannot be blank")
+	}
+
+	ID := uuid.New().String()
+
+	// Check if chat room with the same name exists
+	var existingRoom domain.ChatRoom
+	err := c.dbc.QueryRow("SELECT id, name FROM chatrooms WHERE name = ?", name).Scan(&existingRoom.ID, &existingRoom.Name)
+	if err != nil && err != sql.ErrNoRows {
+		return errors.New("Database error occured")
+	}
+	if existingRoom.Name != "" {
+		return errors.New("Chat room with this name already exists")
+	}
+
+	_, err = c.dbc.Exec("INSERT INTO chatrooms (id, name) VALUES (?, ?)", ID, name)
+	if err != nil {
+		return errors.New("could not create new chatroom")
+	}
+
+	return nil
+}
+
+var jwtKey = []byte(os.Getenv("JWT_SECRET"))
+
+const (
+	tokenDuration     = 60 * time.Minute
+	tokenRefreshLimit = 10 * time.Minute
+)
+
+func extractTokenFromRequest(r *http.Request) string {
+	authHeader := r.Header.Get("Authorization")
+	if authHeader != "" {
+		splitToken := strings.Split(authHeader, "Bearer ")
+		if len(splitToken) == 2 {
+			return splitToken[1]
+		}
+	}
+	return ""
+}
+
+func (c *client) LoginUser(ctx context.Context, email string, password string) (string, error) {
+	var storedUser domain.User
+	if err := c.dbc.QueryRow("SELECT id, name, email, password FROM users WHERE email = ?", email).Scan(&storedUser.ID, &storedUser.Name, &storedUser.Email, &storedUser.Password); err != nil {
+		return "", err
+	}
+
+	// Check password
+	if err := bcrypt.CompareHashAndPassword([]byte(storedUser.Password), []byte(password)); err != nil {
+		return "", err
+	}
+
+	expirationTime := time.Now().Add(tokenDuration)
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+		"id":    storedUser.ID,
+		"name":  storedUser.Name,
+		"email": storedUser.Email,
+		"exp":   expirationTime.Unix(),
+	})
+
+	tokenString, err := token.SignedString(jwtKey)
+	if err != nil {
+		return "", err
+	}
+
+	return tokenString, nil
+}
+
+func tokenVerificationMiddleware(next http.Handler) http.Handler {
```

Comment:
why is this handler inside the database layer?

Reply:
Left here by mistake. Removed now